### PR TITLE
Website Build Pack Changes

### PR DESCRIPTION
**Pull request by SauarvD**
### Requirements

* Read the service metadata to figure out where to deploy the website (account/region)
* Read Jazz configurations to get the details related to the deployment account/region 
  1. AWS Credentials
  2. Origin Access Identity
* Use AWS profile with the correct creds

### Description of the Change

Once the deployment Account and Region are selected and the website service is created, in the Jenkins file, we are fetching the Credential ID by comparing the deployment account id with the Account details coming from the configLoader. Once we get the required Credential ID, we update the s3bucketName, Origin Access Identity and the region from the matching and fetched Account from the configLoader and thus the resources get deployed to the designated Account (and region).

### Benefits

Now User can deploy website service to multiple secondary accounts other than their primary account.

### CODE REVIEW

**Comment by SauarvD** [comment]
> Why deployment_accounts changes are part of this PR?

It is because, the deployment_accounts changes were already committed and pushed to multiaccount branches long ago and the PR was not merged. When I started work on website build pack changes, I took out a fresh branch from multiaccount and committed in that. But the previous changes were still there

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 1]
```diff
@@ -163,10 +172,21 @@ node()  {
 		stage("Deployment to ${current_environment} Env"){
 			def randomString = utilModule.generateRequestId();
 			def credsId = "jazz-${randomString}";
+			def dataObjValues = getRequiredData();
+			def s3BucketNameValue
```
Comment (pspani) [inline builds/jenkins-build-pack-website/Jenkinsfile]: required bucket arn has already been retrieved in line 101/112. Why is it again being fetched?

Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: Removed this line and using the already fetched value

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 2]
```diff
@@ -246,12 +260,22 @@ node()  {
 	}
 }
 
+def getRequiredData(){
```
Comment (SiniWilson) [inline builds/jenkins-build-pack-website/Jenkinsfile]: Can you move this function to util module and make use in all workflows as it is a common method for all, we can avoid the code duplication

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 3]
```diff
@@ -651,12 +684,13 @@ def resetCredentials(credsId) {
 /**
 	Generate the distribution service_config file
 */
-def generateDistributionConfig() {
+def generateDistributionConfig(OAI) {
```
Comment (SiniWilson) [inline builds/jenkins-build-pack-website/Jenkinsfile]: give meaningful name

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 4]
```diff
@@ -651,12 +684,13 @@ def resetCredentials(credsId) {
...
+	def CFOAI = "origin-access-identity/cloudfront/"+OAI
```
Comment (SiniWilson) [inline builds/jenkins-build-pack-website/Jenkinsfile]: give meaningful name

File: builds/jenkins-build-pack-website/Jenkinsfile (hunk 3 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: updated the changes

File: builds/jenkins-build-pack-website/Jenkinsfile (hunk 4 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: updated the changes

File: builds/jenkins-build-pack-website/Jenkinsfile (hunk 2 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: moved the util module

File: builds/jazz-build-module/utility-loader.groovy [hunk 5]
```diff
@@ -171,4 +171,30 @@ def isReplayedBuild() {
   currentBuild.rawBuild.getCauses().any{ cause -> cause.toString().contains(replayClassName) }
 }
 
+/*
+* Getting the required account
+*/
+def getRequiredData(service_config){
```
Comment (pspani) [inline builds/jazz-build-module/utility-loader.groovy]: give a more contextual name to the function

File: builds/jenkins-build-pack-website/Jenkinsfile [hunk 6]
```diff
@@ -454,14 +470,18 @@ def updateBucketPolicy(isCloudfront, existingPolicyObj){
 		def cleanupIndex = -1
 		def CFOAI
```
Comment (pspani) [inline builds/jenkins-build-pack-website/Jenkinsfile]: give meaningful name

File: builds/jazz-build-module/utility-loader.groovy (hunk 5 above)
Comment (SauarvD) [reply builds/jazz-build-module/utility-loader.groovy]: changes the name to getAccountInfo

File: builds/jenkins-build-pack-website/Jenkinsfile (hunk 6 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-website/Jenkinsfile]: changed the name to accessIdentityValue

**Pull request merged**